projects/components/src/table/table.component.ts

```diff
@@ -390,8 +418,9 @@ export class TableComponent
   }
 
   public toggleRowSelection(row: StatefulTableRow): void {
-    this.selection = this.selection === row ? undefined : row;
-    this.selectionChange.emit(this.selection);
+    row.$$state.selected = !row.$$state.selected;
+    this.selections = find(this.selections, row) ? remove(this.selections, row) : this.selections.concat(row);
```

Comment:
can we just use contains (or equivalent) since we're not using the find result other than truthiness